IceCream/Classes/SyncObject.swift

```diff
@@ -23,11 +23,17 @@ public final class SyncObject<T> where T: Object & CKRecordConvertible & CKRecor
     
     public var pipeToEngine: ((_ recordsToStore: [CKRecord], _ recordIDsToDelete: [CKRecord.ID]) -> ())?
     
-    public var realm: Realm
+    public var realm: Realm!
     public var databaseScope: CKDatabase.Scope = .private
     
-    public init(realm: Realm = try! Realm()) {
-      self.realm = realm
+    public init(realm: Realm? = nil) {
+        if realm == nil {
+            BackgroundWorker.shared.start {
+                self.realm = try! Realm()
+            }
+        } else {
+            self.realm = realm!
```

Comment:
Maybe we still have to pass `Realm.Configuration` instead of Realm instance. As it is not a good idea to force outside world to generate Realm instance using `BackgroundWorker`